README.md

```diff
@@ -573,6 +573,49 @@ Don't worry about `adbkey` or `adbkey.pub` under `/.android`, not required.
 
   > Unfortunately it is not possible to pass through a USB device (or a serial port) to a container.
 
+## Firebase Test Lab 
+You can also run your UI tests on Firebase Test Lab using emulators and physical devices, to do so first you need to
+create a service account with an “editor” role in the [Google Cloud Platform](https://console.cloud.google.com/projectselector2/iam-admin/serviceaccounts?pli=1&supportedpurview=project) 
+and enable [Google Cloud Testing API and Google Cloud Results API](https://console.developers.google.com/apis/library?pli=1). Later, in your Google Cloud Console go to “Credentials” > “Create Credentials” > “Create Service Account Key” fill the data and select key type as JSON.
+
+
+* Now let's pull and run the Docker image with Google Cloud SDK
+
+```bash
+# pull the image
+docker pull thyrlian/android-firebase-test-lab
+# run the container passing the service account Json (at least the first time) as external volume
+docker run -d -p 2222:22 -v $(pwd)/sdk:/opt/android-sdk -v /path/on/host/auth.json:/auth.json thyrlian/android-firebase-test-lab
+```
+* Log in and run your tests
+
+```bash
+# check the running process and get a interactive shell
+Docker ps
+Docker exec -it <container-id> /bin/bash
+
+# check Google Cloud is installed properly
+gcloud info --format="value(installation.sdk_root)"
+# authenticate and create any config that you want
+gcloud auth activate-service-account -q --key-file myservicefile.json
+
+# list available devices to run your tests
+gcloud firebase test android models list
+
+# run your tests (make sure you have previosly assemble your app and test apk's)
+APK="--app=debug.apk --test=androidTest.apk"
+TYPE="instrumentation"
+DEVICES="--device model=Pixel2,version=27,locale=en,orientation=portrait" 
+RESULT_DIR="build-result"
+gcloud firebase test android run $APK $DEVICES --type=$TYPE --results-dir=$RESULT_DIR/
```

Comment:
You can remove the appending `/` from `$RESULT_DIR/`.

Reply:
When I run this command, I got an error:

```
ERROR: (gcloud.firebase.test.android.run) No project specified. Please add --project PROJECT_ID to the command line or first run
  $ gcloud config set project PROJECT_ID
```

It seems like you also need to guide users to create a project and config it in [Firebase Console](https://console.firebase.google.com/).

Reply:
BTW, I can't really test this one, Firebase tells me that I can only choose the **Blaze Plan (Pay as you go)** plan but not the **Spark Plan (Free)**. :(

> Since your Google Cloud Platform project has billing enabled, you'll be on the Firebase Blaze plan (pay as you go).

Reply:
oh :( my project is set up with "pay as you go" plan, I thought It could be done with the free plan tbh. 
Regarding --project PROJECT_ID I would probably did it time ago and forgot about it. I'll update the info srry

Reply:
The command says: you need this project option for the first run.

Reply:
Because you use env variables here, to prevent name conflict, I suggest adding prefix like below:

```
UI_TEST_APK="--app=demo-debug.apk --test=demo-debug-androidTest.apk" && UI_TEST_TYPE="instrumentation" && UI_TEST_DEVICES="--device model=Pixel2,version=27,locale=en,orientation=portrait" && UI_TEST_RESULTS_DIR="build-result" && UI_TEST_PROJECT="<your_project_id>"
gcloud firebase test android run $UI_TEST_APK $UI_TEST_DEVICES --type=$UI_TEST_TYPE --results-dir=$UI_TEST_RESULTS_DIR --project=$UI_TEST_PROJECT
```